creator-node/src/components/tracks/TrackTranscodeHandoffManager.js

```diff
@@ -337,36 +340,42 @@ class TranscodeDelegationManager {
     fileNameNoExtension
   }) {
     const originalTrackFormData =
-      await TranscodeDelegationManager.createFormData(fileDir + '/' + fileName)
+      await TrackTranscodeHandoffManager.createFormData(
+        fileDir + '/' + fileName
+      )
 
     // TODO: make this constant. perhaps in a class
     const spID = config.get('spID')
     const { timestamp, signature } =
       generateTimestampAndSignatureForSPVerification(spID, DELEGATE_PRIVATE_KEY)
 
     const resp = await Utils.asyncRetry({
+      logger: TrackTranscodeHandoffManager.logger,
       asyncFn: axios,
-      asyncFnParams: {
-        url: `${sp}/transcode_and_segment`,
-        method: 'post',
-        data: originalTrackFormData,
-        headers: {
-          ...originalTrackFormData.getHeaders()
-        },
-        params: {
-          uuid: fileNameNoExtension,
-          timestamp,
-          signature,
-          spID
-        },
-        adapter: require('axios/lib/adapters/http'),
-        // Set content length headers (only applicable in server/node environments).
-        // See: https://github.com/axios/axios/issues/1362
-        maxContentLength: Infinity,
-        maxBodyLength: Infinity,
-        timeout: SEND_TRANSCODE_AND_SEGMENT_REQUEST_TIMEOUT_MS
-      },
-      asyncFnTask: 'transcode and segment'
+      asyncFnParams: [
```

Comment:
i've now read this a few times and still find the separation of asyncFn and asyncFnParams confusing

esp now that we are passing in an array with single object just to unpack it

we'd get a lot more flexibility and simplicity if we could just pass in
```
asyncFn: (bail) => axios(axiosParams)
```

and this way, per my below coment about moving the back-compat logic out of asyncRetry, this code would then become

```
asyncFn: (bail) => {
  try {
    return axios(axiosReqParams)
  } catch (e) {
    if (<status code = 404) bail()
    throw e
  }
```

Reply:
I thought of passing in the parameters as alleviating boilerplate in using asyncRetry. lets talk about this over chat to come to a conclusion

Reply:
discussed on chat: passing in a fn to replicate og fn